zap/src/main/java/org/parosproxy/paros/extension/history/LogPanel.java

```diff
@@ -253,6 +259,35 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
         return filterButton;
     }
 
+    private JButton getClearButton() {
+        if (clearButton == null) {
+            clearButton = new JButton();
+            clearButton.setIcon(
+                    new ImageIcon(
+                            LogPanel.class.getResource(
+                                    "/resource/icon/fugue/broom-alert.png")));
+            clearButton.setToolTipText(
+                    Constant.messages.getString("history.clear.tooltip"));
+            DisplayUtils.scaleIcon(clearButton);
+
+            clearButton.addActionListener(
+                    new java.awt.event.ActionListener() {
+
+                        @Override
+                        public void actionPerformed(java.awt.event.ActionEvent e) {
+                            if (View.getSingleton()
+                                    .showConfirmDialog(
+                                            Constant.messages
+                                                    .getString("history.clear.confirm")) == JOptionPane.OK_OPTION) {
```

Comment:
I don't think this is necessary, the `purgeHistory` already confirms the deletion.

Reply:
I tried removing the if-condition, but it didn't open up any dialog. Am I missing something?